tests/test_token.py

```diff
@@ -96,3 +96,17 @@ def token_auth_route2():
         self.assertTrue('WWW-Authenticate' in response.headers)
         self.assertEqual(response.headers['WWW-Authenticate'],
                          'Token realm="foo"')
+
+    def test_token_auth_custom_header_valid_token(self):
+        self.token_auth.header = 'X-API-Key'
+        response = self.client.get(
+            '/protected', headers={'X-API-Key': 'MyToken this-is-the-token!'})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data.decode('utf-8'), 'token_auth:user')
+
+    def test_token_auth_custom_header_invalid_token(self):
+        self.token_auth.header = 'X-API-Key'
+        response = self.client.get(
+            '/protected', headers={'X-API-Key': 'invalid-token-should-fail'})
+        self.assertEqual(response.status_code, 401)
+        self.assertTrue('WWW-Authenticate' in response.headers)
```

Comment:
Add a test for token auth that includes a correct `Authorization` header for basic authentication. In that case we'd want the basic auth to be ignored in favor of the custom token auth.

Reply:
Good point, just did `HTTPAuthTestCase::test_token_auth_custom_header_over_default_token`